### PR TITLE
samples: zbus: add confirmed channel sample

### PR DESCRIPTION
**Pull request by rodrigopex**
This commit adds a new sample to zbus that shows how to implement a confirmed channel. Developers commonly asked about similar features on zbus discord channel and RFC (in the past). Adding this sample, we can show developers how to solve their needs (confirmed channels) without zbus changes.

### CODE REVIEW

File: samples/subsys/zbus/confirmed_channel/src/main.c [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+ZBUS_SUBSCRIBER_DEFINE(bar_sub3, 4);
+static void bar_sub3_task(void)
+{
+	const struct zbus_channel *chan;
+
+	while (!zbus_sub_wait(&bar_sub3, &chan, K_FOREVER)) {
+		struct confirmed_msg cm;
+
+		if (&confirmed_chan == chan) {
```
Comment (fabiobaltieri) [inline samples/subsys/zbus/confirmed_channel/src/main.c:91]: Same as above, swap the logic.

File: samples/subsys/zbus/confirmed_channel/src/main.c [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+ZBUS_SUBSCRIBER_DEFINE(bar_sub1, 4);
+
+static void bar_sub1_task(void)
+{
+	const struct zbus_channel *chan;
+
+	while (!zbus_sub_wait(&bar_sub1, &chan, K_FOREVER)) {
+		struct confirmed_msg cm;
+
+		if (&confirmed_chan == chan) {
```
Comment (fabiobaltieri) [inline samples/subsys/zbus/confirmed_channel/src/main.c:44]: Swap the logic (see my comment below)

**Review by fabiobaltieri — DISMISSED** [review]

File: samples/subsys/zbus/confirmed_channel/src/main.c [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+ * Copyright (c) 2023 Rodrigo Peixoto <rodrigopex@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/zbus/zbus.h>
+LOG_MODULE_DECLARE(zbus, CONFIG_ZBUS_LOG_LEVEL);
+
+atomic_t sub_count = ATOMIC_INIT(0);
```
Comment (fabiobaltieri) [inline samples/subsys/zbus/confirmed_channel/src/main.c:11]: static here as well

**Review by fabiobaltieri — APPROVED** [review]

**Review by rodrigopex — COMMENTED** [review]
I have fixed the code based on the requested changes.

**Review by cfriedt — APPROVED** [review]

**Pull request merged**